src/main/scala/shine/DPIA/FunctionalPrimitives/Iterate.scala

```diff
@@ -47,7 +48,7 @@ final case class Iterate(n: Nat,
   override def xmlPrinter: Elem = {
     val l = f.t.x
     <iterate n={ToString(n)} m={ToString(m)} k={ToString(k)} dt={ToString(dt)}>
-      <f type={ToString(l ->: ExpType(ArrayType(l, dt), read) ->: ExpType(ArrayType(l /^ n, dt), read))}>
+      <f type={ToString(l ~>: ExpType(ArrayType(l, dt), read) ->: ExpType(ArrayType(l /^ n, dt), read))}>
```

Comment:
Should this not be a type constructing arrow `->:`?

Reply:
Yeah, it should be. I used intelliJ refactoring for some parts and it messed things up :/
Let me know if you find other mistakes like this.

Reply:
Although ideally we would not need to implement this xmlPrinter manually.

Reply:
There are quite a few more cases of this ...